numpy/core/src/npysort/quicksort.c.src

```diff
@@ -107,6 +111,7 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
                 *sptr++ = pi - 1;
                 pl = pi + 1;
             }
+            *psdepth++ = cdepth + 1;
```

Comment:
Maybe I'm missing something looking at the diff. But each push should have a depth increase so that depth is monotone up the stack. Maybe you increase it somewhere else also?